src/libaktualizr/utilities/utils.cc

```diff
@@ -883,8 +883,13 @@ int Socket::connect() {
   return ::connect(socket_fd_, reinterpret_cast<const struct sockaddr *>(&sock_address_), sizeof(sock_address_));
 }
 
+std::mutex CurlEasyWrapper::curl_init_mutex;
+
 CurlEasyWrapper::CurlEasyWrapper() {
-  handle = curl_easy_init();
+  {
```

Comment:
It could also be avoided if curl_global_init is called.
In the worst case, the mutex could also be removed from the class definition and just declared as static in the method.